### PR TITLE
I made a jukebox

### LINKED ISSUE #178: [Stretch Goal] Jukebox
Just pick a song from another mode and play it

### PR DESCRIPTION
**Pull request by Brycey92**
It's implemented!

This closes #178

### CODE REVIEW

File: main/modes/mode_credits.c [hunk 1]
```diff
@@ -70,6 +70,7 @@ static const char* creditNames[] =
     "Dac\n",
     "Decnav\n",
     "dylwhich\n",
+    "Ellie (VanillyNeko.com)\n",
```
Comment (AEFeinstein) [inline main/modes/mode_credits.c:73]: Haha, I was gonna nudge you to fill out the form, but that works too!

File: main/modes/fighter/mode_fighter.c [hunk 2]
```diff
@@ -312,7 +312,7 @@ static const song_t f1hit =
 };
 
 // Simple sound effect when fighter 2 is hit
-song_t f2hit =
+const song_t f2hit =
```
Comment (AEFeinstein) [inline main/modes/fighter/mode_fighter.c:315]: I think you missed `f1hit`

File: main/modes/mode_jukebox.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+#ifndef _MODE_JUKEBOX_H_
+#define _MODE_JUKEBOX_H_
+
+#include "musical_buzzer.h" 
+
+extern swadgeMode modeJukebox;
+
+void setJukeboxMainMenu(void);
```
Comment (AEFeinstein) [inline main/modes/mode_jukebox.h:15]: This doesn't _need_ to be in the header

**Comment by AEFeinstein** [comment]
There's enough space that you can put a little more margin between the corners and the text. The pixels shouldn't be touching the curved boundary.
image

File: main/modes/mode_jukebox.c [hunk 4]
```diff
@@ -0,0 +1,750 @@
...
+
+    jukebox->disp = disp;
+
+    loadFont("ibm_vga8.font", &jukebox->ibm_vga8);
+    loadFont("radiostars.font", &jukebox->radiostars);
+    loadFont("mm.font", &jukebox->mm);
+
+    loadWsg("arrow12.wsg", &jukebox->arrow);
+
+    jukebox->menu = initMeleeMenu(str_jukebox, &(jukebox->mm), jukeboxMainMenuCb);
```
Comment (AEFeinstein) [inline main/modes/mode_jukebox.c:309]: Don't forget to call `deinitMeleeMenu()`!

File: main/modes/mode_jukebox.c [hunk 5]
```diff
@@ -0,0 +1,750 @@
...
+        {
+            drawMeleeMenu(jukebox->disp, jukebox->menu);
+            break;
+        }
+        case JUKEBOX_PLAYER:
+        {
+            jukeboxLedDances[jukebox->danceIdx].func(elapsedUs, jukeboxLedDances[jukebox->danceIdx].arg, jukebox->resetDance);
+            jukebox->resetDance = false;
+
+            fillDisplayArea(jukebox->disp, 0, 0, jukebox->disp->w, jukebox->disp->h, c010);
```
Comment (AEFeinstein) [inline main/modes/mode_jukebox.c:483]: Filling the whole area after calling `clearPx()` is a bit redundant. You can move `clearPx()` to just the `JUKEBOX_MENU` case.

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

**Comment by AEFeinstein** [comment]
Aside from the requested changes, I'll keep an eye out for new music and integrate it as appropriate.

**Comment by Brycey92** [comment]
> There's enough space that you can put a little more margin between the corners and the text. The pixels shouldn't be touching the curved boundary. image

this is the same setting as in tunernome and slide whistle, I can change it in those too

File: main/modes/mode_jukebox.c (hunk 5 above)
Comment (Brycey92) [reply main/modes/mode_jukebox.c:483]: done

File: main/modes/mode_jukebox.c (hunk 4 above)
Comment (Brycey92) [reply main/modes/mode_jukebox.c:309]: done

File: main/modes/mode_jukebox.h (hunk 3 above)
Comment (Brycey92) [reply main/modes/mode_jukebox.h:15]: moved to C file

File: main/modes/fighter/mode_fighter.c (hunk 2 above)
Comment (Brycey92) [reply main/modes/fighter/mode_fighter.c:315]: both appear to have been missed - added them!

**Comment by Brycey92** [comment]
fixed the pixels touching the curved boundary in all 3 modes

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**